components/codec/src/number.rs

```diff
@@ -481,19 +483,57 @@ impl NumberCodec {
     pub fn try_decode_var_i64(buf: &[u8]) -> Result<(i64, usize)> {
         let (uv, decoded_bytes) = Self::try_decode_var_u64(buf)?;
         let v = uv >> 1;
-        if uv & 1 == 0 {
+        if unsafe { likely(uv & 1 == 0) } {
```

Comment:
I don't think the `likely` here has any effect 🤔.

https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=8781de9fda4b87a4d630d94f9d53a894

Reply:
Nice catch. I am too lazy to check if all `likely` are effective. So there will be actually no jump instructions in this case.